oxenstored/connections.ml

```diff
@@ -57,23 +66,36 @@ let add_domain cons dom =
     )
   in
   let con = Connection.create xbcon (Some dom) in
-  Hashtbl.add cons.domains (Domain.get_id dom) con ;
-  Hashtbl.add cons.ports (Domain.get_local_port dom) con
-
-let select ?(only_if = fun _ -> true) cons =
-  Hashtbl.fold
-    (fun _ con (ins, outs) ->
-      if only_if con then
-        let fd = Connection.get_fd con in
-        let in_fds = if Connection.can_input con then fd :: ins else ins in
-        let out_fds = if Connection.has_output con then fd :: outs else outs in
-        (in_fds, out_fds)
-      else
-        (ins, outs)
+  Hashtbl.replace cons.domains (Domain.get_id dom) con ;
+  Hashtbl.replace cons.ports (Domain.get_local_port dom) con
+
+let refresh_poll_status ?(only_if = fun _ -> true) cons spec_fds =
+  (* special fds are always read=true, but get overwritten by select_stubs, so we
+     need to reset the event we are polling for *)
+  let _ =
```

Comment:
Minor: the `let _` is not needed here if you use `List.iteri`, the result is a unit (it is better to remove it, so that you'd get a warning/build error if you accidentally partially apply the iter)